runtime/src/main/cpp/Memory.cpp

```diff
@@ -689,6 +698,31 @@ inline container_size_t objectSize(const ObjHeader* obj) {
   return alignUp(size, kObjectAlignment);
 }
 
+template <typename func>
+inline void traverseObjectFields(ObjHeader* obj, func process) {
```

Comment:
The entire implementation of this function is quite similar to `traverseContainerObjectFields `. Is it possible to extract the common part?